### ISSUE TITLE
Incorrect error for type already defined in scope using raw identifiers

### ISSUE THREAD

**Issue by olegnn** [rust-lang/rust#69054]
This code

```rust
trait T<r#fn> {
    fn f<r#fn>() {}
}

fn main() {}

```

([Playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=73f56a1b4d4ac37801c60b8e8738242d))

Gives incorrect error for type declared using raw identifiers:

```
error[E0403]: the name `fn` is already used for a generic parameter in this item's generic parameters
 --> src/main.rs:2:10
  |
1 | trait T<r#fn> {
  |         ---- first use of `fn`
2 |     fn f<r#fn>() {}
  |          ^^^^ already used

warning: type parameter `fn` should have an upper camel case name
 --> src/main.rs:1:9
  |
1 | trait T<r#fn> {
  |         ^^^^ help: convert the identifier to upper camel case: `Fn`
  |
  = note: `#[warn(non_camel_case_types)]` on by default

warning: type parameter `fn` should have an upper camel case name
 --> src/main.rs:2:10
  |
2 |     fn f<r#fn>() {}
  |          ^^^^ help: convert the identifier to upper camel case: `Fn`

error: aborting due to previous error

For more information about this error, try `rustc --explain E0403`.
error: could not compile `playground`.
```

**Comment by krnik** [comment]
Isn't this error correct by any chance? You defined two generic parameters with conflicting names. 

What is the expected error message in such cases?

**Comment by estebank** [comment]
Current output:

```
error[E0403]: the name `r#fn` is already used for a generic parameter in this item's generic parameters
 --> src/main.rs:2:10
  |
1 | trait T<r#fn> {
  |         ---- first use of `r#fn`
2 |     fn f<r#fn>() {}
  |          ^^^^ already used

warning: type parameter `fn` should have an upper camel case name
 --> src/main.rs:1:9
  |
1 | trait T<r#fn> {
  |         ^^^^ help: convert the identifier to upper camel case: `Fn`
  |
  = note: `#[warn(non_camel_case_types)]` on by default

warning: type parameter `fn` should have an upper camel case name
 --> src/main.rs:2:10
  |
2 |     fn f<r#fn>() {}
  |          ^^^^ help: convert the identifier to upper camel case: `Fn`
```
